### PR TITLE
export ESM modules with rollup

### PR DESCRIPTION
**Pull request by AGalabov**
Alternative approach to ESM modules using Rollup as dev dependency

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-        node-version: [14.x, 16.x, 18.x]
```
Comment (AGalabov) [inline .github/workflows/check.yml:17]: Rollup uses `??=` internally which is not valid in node 14. I don't like the fact that we had to do this change but at the same time I do not think that it is too much of a problem :thinking: .

**Comment by danielkarso-pon** [comment]
This PR looks good. I would prefer this over my PR!

File: .github/workflows/check.yml (hunk 1 above)
Comment (georgyangelov) [reply .github/workflows/check.yml:17]: That's weird, are you sure it's not configurable?

Comment (AGalabov) [reply .github/workflows/check.yml:17]: Well that's rollup's [source code](https://github.com/rollup/rollup/blob/master/src/utils/crypto.ts#L21) :thinking: I don't think we can configure that (at least I did not find any options and don't think there are). If we wanted to we probably can introduce Babel and try and convert it to old syntax ... not sure if there's any point.

**Pull request merged**